### PR TITLE
[R4R]: Adjust get block range logic for proposer;add checks for minimum bloc…

### PR DESCRIPTION
**Pull request by byteflyfunny**
# Goals of PR

Core changes:

- Adjust get block range logic for proposer
- Add checks for minimum block number and timeout

### CODE REVIEW

File: batch-submitter/drivers/proposer/driver.go [hunk 1]
```diff
@@ -33,7 +33,6 @@ import (
 	rollupTypes "github.com/mantlenetworkio/mantle/fraud-proof/rollup/types"
 	l2types "github.com/mantlenetworkio/mantle/l2geth/core/types"
 	l2ethclient "github.com/mantlenetworkio/mantle/l2geth/ethclient"
-	"github.com/mantlenetworkio/mantle/l2geth/rollup"
```
Comment (wukongcheng) [inline batch-submitter/drivers/proposer/driver.go:36]: Please adjust the grouping of imports according to the code style guidelines.

**Review by wukongcheng — CHANGES_REQUESTED** [review]

File: batch-submitter/drivers/proposer/driver.go [hunk 2]
```diff
@@ -219,12 +218,20 @@ func (d *Driver) GetBatchBlockRange(
...
 
-	backend, _ := rollup.NewBackend("l1")
-	index, err := d.cfg.RollClient.GetLatestTransactionIndex(backend)
+	currentHeader, err := d.cfg.L1Client.HeaderByNumber(ctx, nil)
 	if err != nil {
 		return nil, nil, err
 	}
-	end := new(big.Int).SetUint64(*index)
+	finality := new(big.Int).SetUint64(d.cfg.FinalityConfirmations)
+	finality.Add(finality, new(big.Int).SetInt64(2)) // add 2 block number buffer to dtl sync data
```
Comment (guoshijiang) [inline batch-submitter/drivers/proposer/driver.go:226]: This `2` buffer recommended to define it as a constant

File: batch-submitter/drivers/proposer/driver.go (hunk 1 above)
Comment (byteflyfunny) [reply batch-submitter/drivers/proposer/driver.go:36]: done

File: batch-submitter/drivers/proposer/driver.go (hunk 2 above)
Comment (byteflyfunny) [reply batch-submitter/drivers/proposer/driver.go]: done

**Review by guoshijiang — APPROVED** [review]

**Review by guoshijiang — APPROVED** [review]

**Review by bradyjoestar — APPROVED** [review]

**Review by guoshijiang — APPROVED** [review]

**Review by Sha3nS — APPROVED** [review]

**Review by wukongcheng — APPROVED** [review]

**Pull request merged**